pkg/azure/access/helpers/vm.go

```diff
@@ -0,0 +1,224 @@
+	defer cancelFn()
+	poller, err := vmAccess.BeginDelete(delCtx, resourceGroup, vmName, nil)
+	if err != nil {
+		errors.LogAzAPIError(err, "Failed to trigger delete of VM [ResourceGroup: %s, VMName: %s]", resourceGroup, vmName)
+		return
+	}
+	_, err = poller.PollUntilDone(delCtx, nil)
+	if err != nil {
+		errors.LogAzAPIError(err, "Polling failed while waiting for delete of VM: %s for ResourceGroup: %s", vmName, resourceGroup)
+		return
+	}
+	return
+}
+
+// CreateVirtualMachine creates a Virtual Machine given a resourceGroup and virtual machine creation parameters.
+// NOTE: All calls to this Azure API are instrumented as prometheus metric.
+func CreateVirtualMachine(ctx context.Context, vmAccess *armcompute.VirtualMachinesClient, resourceGroup string, vmCreationParams armcompute.VirtualMachine) (vm *armcompute.VirtualMachine, err error) {
+	defer instrument.RecordAzAPIMetric(err, vmCreateServiceLabel, time.Now())
+	createCtx, cancelFn := context.WithTimeout(ctx, defaultCreateVMTimeout)
+	defer cancelFn()
+	vmName := *vmCreationParams.Name
+	poller, err := vmAccess.BeginCreateOrUpdate(createCtx, resourceGroup, vmName, vmCreationParams, nil)
+	if err != nil {
+		errors.LogAzAPIError(err, "Failed to trigger create of VM [ResourceGroup: %s, VMName: %s]", resourceGroup, vmName)
+		return
+	}
+	createResp, err := poller.PollUntilDone(createCtx, nil)
+	if err != nil {
+		errors.LogAzAPIError(err, "Polling failed while waiting for create of VM: %s for ResourceGroup: %s", vmName, resourceGroup)
+		return
+	}
+	vm = &createResp.VirtualMachine
+	return
+}
+
+// SetCascadeDeleteForNICsAndDisks sets cascade deletion for NICs and Disks (OSDisk and DataDisks) associated to passed virtual machine.
+// NOTE: All calls to this Azure API are instrumented as prometheus metric.
+func SetCascadeDeleteForNICsAndDisks(ctx context.Context, vmClient *armcompute.VirtualMachinesClient, resourceGroup string, vm *armcompute.VirtualMachine) (err error) {
+	defer instrument.RecordAzAPIMetric(err, vmUpdateServiceLabel, time.Now())
+	vmUpdateParams := createVMUpdateParamsForCascadeDeleteOptions(vm) // TODO: Rename this method it returns a VM not "params"
+	if vmUpdateParams == nil {
+		klog.Infof("All configured NICs, OSDisk and DataDisks have cascade delete already set. Skipping update of VM: [ResourceGroup: %s, Name: %s]", resourceGroup, *vm.Name)
+		return
+	}
+	updCtx, cancelFn := context.WithTimeout(ctx, defaultUpdateVMTimeout)
+	defer cancelFn()
+	poller, err := vmClient.BeginUpdate(updCtx, resourceGroup, *vm.Name, *vmUpdateParams, nil)
+	if err != nil {
+		errors.LogAzAPIError(err, "Failed to trigger update of VM [ResourceGroup: %s, VMName: %s]", resourceGroup, *vm.Name)
+		return
+	}
+	pollResp, err := poller.PollUntilDone(updCtx, nil)
+	if err != nil {
+		errors.LogAzAPIError(err, "Polling failed while waiting for update of VM: %s for ResourceGroup: %s", *vm.Name, resourceGroup)
+		return
+	}
+
+	_, err = pollResp.MarshalJSON()
```

Comment:
could you help us know why this is needed? aren't we getting err already from `PollUntilDone` call above?

Reply:
I think this was added as a 'safety' check, but probably can be removed.

Reply:
Removed. PTAL